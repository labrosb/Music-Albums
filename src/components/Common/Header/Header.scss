@import '../../../styles/colors.scss';
@import '../../../styles/mixins.scss';

.header {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 90px;
  padding: 0 78px;
  color: color(primary-contrast);
  background-color: color(primary);
  .back-button {
    cursor: pointer;
  }
  .icon {
    font-size: 20px;
    color: color(background);
  }
  .title {
    display: flex;
    align-items: center;
    padding-left: 200px;
    margin: auto;
    .title-label {
      @include big-label(background, center)
    }
    .icon {
      font-size: 18px;
      margin-bottom: 8px;
      margin-left: 10px;
    }
  }
  .menu {
    display: flex;
    justify-content: space-between;
    width: 190px;
    .route {
      @include text(normal, background);
      cursor: pointer;
    }
    .route.active::before {
      content: "âžŸ ";
      line-height: 0;
    }
  }
}

@media (max-width: 900px) {
  .header {
    justify-content: space-between;
    padding: 0 7%;
    .title {
      padding-left: 0;
      margin: 0;
    }
  }
}

@media (max-width: 420px) {
  .header {
    padding: 0 5%;
    .icon {
      margin-left: 7px;
      font-size: 14px;
    }
    .title {
      .title-label {
        @include med-label(background, center)
      }
      .icon {
        margin-bottom: 7px;
        margin-left: 7px;
        font-size: 14px;
      }
    }
    .menu {
      width: 160px;
      .route {
        @include text(small, background);
      }
    }
  }
}
