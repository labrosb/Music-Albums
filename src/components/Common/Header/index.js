import React, { useCallback } from 'react';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faMusic, faArrowLeft } from '@fortawesome/free-solid-svg-icons';
import './Header.scss';
/**
 * App Header component
 * @component
 * @param {string} header - The title of the header
 * @param {string} active - The active link / root in the header
 * @param {boolean} noBackButton - Indicates if header has back button
 */
function Header({ title, active, noBackButton }) {
  const history = useHistory();
  /** Function called on Top 100 link click */
  const redirectTop100 = useCallback(() => {
    history.push({ pathname: '/top-100' });
  }, [history]);

  /** Function called on Favorites link click */
  const redirectToFavorites = useCallback(() => {
    history.push({ pathname: '/my-favorites' });
  }, [history]);

  /** Function called on back arrow click */
  const goBack = useCallback(() => {
    history.goBack();
  }, [history]);

  /** Function conditionally renders back button */
  const renderBackButton = () => {
    if (!noBackButton) {
      return (
        <div
          className="back-button"
          onClick={goBack}
        >
          <FontAwesomeIcon
            className="icon"
            icon={faArrowLeft}
          />
        </div>
      );
    }
    return null;
  };

  return (
    <div className="header">
      {renderBackButton()}
      <div className="title">
        <h1 className="title-label">
          {title || 'Music'}
        </h1>
        <FontAwesomeIcon
          className="icon"
          icon={faMusic}
        />
      </div>
      <div className="menu">
        <span
          className={`route ${active === 'top100' ? 'active' : ''}`}
          onClick={redirectTop100}
        >
          Top 100
        </span>
        <span
          className={`route ${active === 'favorites' ? 'active' : ''}`}
          onClick={redirectToFavorites}
        >
          My Favorites
        </span>
      </div>
    </div>
  );
}

Header.propTypes = {
  title: PropTypes.string,
  active: PropTypes.string,
  noBackButton: PropTypes.bool
};

export default Header;
