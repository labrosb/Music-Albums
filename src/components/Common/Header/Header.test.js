/* eslint-disable no-undef */
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Header from './index';

configure({ adapter: new Adapter() });

jest.mock('react-router-dom', () => ({
  useHistory: jest.fn()
}));

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Header title="My Header" active="top100" />);
});

describe('Header component', () => {
  it('Renders Header Title passed via props correctly', () => {
    const title = wrapper.find('.title-label').text();

    expect(title).toEqual('My Header');
  });

  it('Has active (selected) route element', () => {
    const route = wrapper.find('.route').first();

    expect((route).hasClass('active')).toBe(true);
  });

  it('Matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
