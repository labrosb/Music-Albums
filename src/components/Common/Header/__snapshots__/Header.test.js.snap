// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Matches snapshot 1`] = `
<div
  className="header"
>
  <div
    className="back-button"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="icon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f060",
            "M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z",
          ],
          "iconName": "arrow-left",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <div
    className="title"
  >
    <h1
      className="title-label"
    >
      My Header
    </h1>
    <FontAwesomeIcon
      border={false}
      className="icon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f001",
            "M470.38 1.51L150.41 96A32 32 0 0 0 128 126.51v261.41A139 139 0 0 0 96 384c-53 0-96 28.66-96 64s43 64 96 64 96-28.66 96-64V214.32l256-75v184.61a138.4 138.4 0 0 0-32-3.93c-53 0-96 28.66-96 64s43 64 96 64 96-28.65 96-64V32a32 32 0 0 0-41.62-30.49z",
          ],
          "iconName": "music",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <div
    className="menu"
  >
    <span
      className="route active"
      onClick={[Function]}
    >
      Top 100
    </span>
    <span
      className="route "
      onClick={[Function]}
    >
      My Favorites
    </span>
  </div>
</div>
`;
